lib/puppet/type/grub_config.rb

```diff
@@ -45,6 +45,18 @@
     desc <<-EOM
       Value of the GRUB parameter.
     EOM
+
+    munge do |value|
+      value.to_s unless [Hash, Array].include?(value.class)
+    end
+
+    def insync?(is)
+      if is.is_a?(String) && should.is_a?(String)
```

Comment:
At some point, we should probably rewrite `augeasproviders` with the new Resource API to natively support data types.

Reply:
I'd be curious to see how that works out. From what I've seen, easy things are easy and advanced things are nigh impossible (might be missing something though).

Reply:
That was my feeling as well. @DavidS assured me otherwise though…

Reply:
that severely depends on your definition of "advanced".

Reply:
looking over the `grub_config` type and provider, this autorequire computation seems to be the one thing that the Resource API can't handle: https://github.com/hercules-team/augeasproviders_grub/blob/402ee21e8a590ed2c4e0c3981e35c7a03d268c61/lib/puppet/type/grub_config.rb#L63-L85

Reply:
The stuff we do in https://github.com/hercules-team/augeasproviders_core/blob/master/lib/puppet/provider/augeasprovider/default.rb for ex, such as the handler sharing in the Puppet session.

`instances` such as https://github.com/hercules-team/augeasproviders_ssh/blob/master/lib/puppet/provider/sshd_config/augeas.rb#L113

probably other things, too. @trevor-vaughan do you have things in mind?

Reply:
The `instances` method seems to be fine - that's a regular `get`, collecting and computing all the resources.

I didn't fully read through the handler sharing, but you should be able to share any instances you need across providers by directly reaching into other ruby namespaces. No need to load something into the catalog for that.